### PR TITLE
Stabilise key dependencies

### PR DESCRIPTION
**Pull request by luckysori**
Fixes https://github.com/get10101/10101/issues/766 (they pass locally :D).
Fixes https://github.com/get10101/10101/issues/768.
Fixes https://github.com/get10101/10101/issues/664.
Fixes https://github.com/get10101/10101/issues/663 (we don't run into it and the app gets their money back [^1]).

The main motivation is to stay on top of the latest changes to `rust-lightning` and `rust-dlc`, whilst also definitively removing some of the things that we temporarily added during our latest bug hunt.

Some patches were also removed which might cause some controversy:

- https://github.com/get10101/rust-dlc/commit/ce78c25.
- https://github.com/get10101/rust-dlc/commit/cf3d1eb.
- https://github.com/get10101/rust-dlc/commit/304e85d.

In general reviewers are encouraged to take a look at the differences between these branches:

- For `rust-dlc` we are now using https://github.com/get10101/rust-dlc/commit/8fa3692 [^2] instead of https://github.com/get10101/rust-dlc/commit/2322288.

- For `rust-lightning` we are now using https://github.com/get10101/rust-lightning/commit/c65cb1c5 [^3] instead of https://github.com/get10101/rust-lightning/commit/939ebb43.

The differences are not so obvious because we diverged quite a bit. We still want to keep certain commits that haven't been upstreamed yet.

Other notable changes
---------------------

- Adapting to the new API to force-close the LN-DLC channel provided by `rust-dlc`. As such, we've modified some code around that and updated the `force_close_ln_dlc_channel` test accordingly.

- Introducing `wait_until_dlc_channel_state` to replace a lot of eager sleeps which led to slower and flakier tests.

- Enabling the configuration of the `dlc_manager_periodic_check_interval` and `sub_channel_manager_periodic_check_interval` viat the `LnDlcNodeSettings`. Originally this was added to be able to circumvent another issue, but the issue has now been fixed in
`rust-dlc`.

- Ensuring that we drop the `LnDlcNodeSettings` read lock as soon as possible so that we don't hold onto it during the sleeps of certain periodic tasks. This fixes a bug as in a previous version of this patch we were not able to update the settings in the tests, because there was always at least one read lock being held!

[^1]: The test can still be improved by making sure that the oracle unlocks a CET that pays to both parties. Atm we use an event whose attestation results in the coordinator getting the whole DLC output. This means that we are only verifying that the part of the LN channel that belongs to the app is returned.
[^2]: This branch is temporarily named `feature/ln-dlc-channels-114-tibo`.
[^3]: This branch is temporarily named `split-tx-experiment-114-tibo`.

### CODE REVIEW

**Comment by holzeis** [comment]
> https://github.com/get10101/rust-dlc/commit/cf3d1eb.
> https://github.com/get10101/rust-dlc/commit/304e85d.

I am not against reverting these changes, but what does that mean for our existing production coordinator? Do we need to clear the dlc db for the mainnet coordiantor to work with these changes?

**Comment by holzeis** [comment]
> Fixes https://github.com/get10101/10101/issues/766 (they pass locally :D).

Since the tests are not passing in the CI, I guess your changes are not fixing but only improving #766?

File: crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs
Comment (holzeis) [inline crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs:69]: 🤔 Can we be sure here, that the coordinator hasn't processed the accept message before the reconnect?

**Comment by luckysori** [comment]
> > In general reviewers are encouraged to take a look at the differences between these branches:
> > 
> > * For rust-dlc we are now using [get10101/rust-dlc@8fa3692](https://github.com/get10101/rust-dlc/commit/8fa3692) [2](https://github.com/get10101/10101/pull/771#user-content-fn-2-bf02b6925d0d4225f01997aa004b35f1) instead of [get10101/rust-dlc@2322288](https://github.com/get10101/rust-dlc/commit/2322288).
> > * For rust-lightning we are now using [get10101/rust-lightning@c65cb1c5](https://github.com/get10101/rust-lightning/commit/c65cb1c5) [3](https://github.com/get10101/10101/pull/771#user-content-fn-3-bf02b6925d0d4225f01997aa004b35f1) instead of [get10101/rust-lightning@939ebb43](https://github.com/get10101/rust-lightning/commit/939ebb43).
> 
> The referenced change seem insignificant, unless the real changes are hiding in the changed dependency to our get10101 fork? Am I missing something?

The linked patches are just heads of branches. The branches themselves are referenced in the footnotes and also if you follow the links:

image

**Comment by luckysori** [comment]
> > Fixes #766 (they pass locally :D).
> 
> Since the tests are not passing in the CI, I guess your changes are not fixing but only improving #766?

Yeah, at the moment we appear to still have at least one other flaky test. I will try to fix it tomorrow.

**Review by holzeis — APPROVED** [review]
LGTM 👍 

However, I did not review all changes implicit through the upgraded dependencies.

**Comment by holzeis** [comment]
@luckysori https://github.com/get10101/rust-lightning/commit/6b059a7222997015141fb9aa44788f4a43732957#commitcomment-118021986

File: crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs
Comment (luckysori) [reply crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs:61]: I honestly would have removed this reconnect from the original test. I didn't quite understand the purpose.

File: crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs
Comment (luckysori) [reply crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs:69]: We can be sure in the sense that nothing happens unless we call `process_incoming_messages` (bad name because it also sends messages that are queued up) implicitly through `wait_until_dlc_channel_state` or `periodic_check` on the `SubChannelManager` implicitly through `sub_channel_manager_periodic_check`.

But we can certainly check that the message has not arrived yet, by asserting on the state of the app (should be `Offered`?) or maybe checking the coordinator's message queue manually.

File: crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs
Comment (luckysori) [reply crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs:72]: > An assertion that a message is pending would be nice here. Otherwise we are just implying that this will resend the accept message

Covered with my response above.

> Also shouldn't it be the confirm message? The accept message is sent by the app.

We never adapted these tests, so we are not fully mimicking what we do in production. That is, the app proposes the channel open and close in most of these tests (if not all).

**Comment by luckysori** [comment]
I've merged #772 and #773 into this PR after getting approvals there.

File: crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs
Comment (holzeis) [reply crates/ln-dlc-node/src/tests/dlc/dlc_setup_with_reconnects.rs:69]: so isn't there a background task running automatically trying to process any incoming messages? If so, I think the asserting on the state should be fine. But, I think the `SubChannelState` of the app should be `Confirmed` after the coordinator processes the accept dlc channel offer?

btw. I noticed that in the tests we have the app and the coordinator roles inverted. But I guess that shouldn't have an implications.

**Comment by luckysori** [comment]
> > > Fixes #766 (they pass locally :D).
> > 
> > 
> > Since the tests are not passing in the CI, I guess your changes are not fixing but only improving #766?
> 
> Yeah, at the moment we appear to still have at least one other flaky test. I will try to fix it tomorrow.

@holzeis: I've opened #774 and #775 to capture the two tests that are still flaky. The one related to the JIT payment is way more likely to happen as we could have probably guessed. I think we are safe to close #766 with this PR since it was too general anyway and it has fixed other tests that were intermittently failing because we used static sleeps.

**Comment by luckysori** [comment]
> > > [get10101/rust-dlc@cf3d1eb](https://github.com/get10101/rust-dlc/commit/cf3d1eb).
> > > [get10101/rust-dlc@304e85d](https://github.com/get10101/rust-dlc/commit/304e85d).
> > 
> > 
> > I am not against reverting these changes, but what does that mean for our existing production coordinator? Do we need to clear the dlc db for the mainnet coordiantor to work with these changes?
> 
> I think before upgrading we should force-close all the LN-DLC channels that we have. And then purge the DLC database.

I've introduced https://github.com/p2pderivatives/rust-dlc/commit/3b277dcd7d573eac615fe2cecf0325c45413bcb9 which _should_ allow us to update mainnet without closing the problematic positions.

**Pull request merged**
